tardis/montecarlo/src/cmontecarlo.c

```diff
@@ -829,41 +832,56 @@ montecarlo_main_loop(storage_model_t * storage, int64_t virtual_packet_flag, int
 #ifdef WITHOPENMP
   omp_set_dynamic(0);
   omp_set_num_threads(nthreads);
-#pragma omp parallel
+
+#pragma omp parallel firstprivate(finished_packets)
   {
     rk_state mt_state;
     rk_seed (seed + omp_get_thread_num(), &mt_state);
 #pragma omp master
     fprintf(stderr, "Running with OpenMP - %d threads\n", omp_get_num_threads());
-
 #pragma omp for
 #else
-  fprintf(stderr, "Running without OpenMP\n");
-  rk_state mt_state;
-  rk_seed (seed, &mt_state);
+    rk_state mt_state;
+    rk_seed (seed, &mt_state);
+    fprintf(stderr, "Running without OpenMP\n");
 #endif
-  for (int64_t packet_index = 0; packet_index < storage->no_of_packets; packet_index++)
+    for (int64_t packet_index = 0; packet_index < storage->no_of_packets; packet_index++)
     {
+      ++finished_packets;
+      if ( finished_packets%100 == 0 ) {
+#ifdef WITHOPENMP
+        // WARNING: This only works with a static sheduler and gives an approximation of progress.
+        // The alternative would be to have a shared variable but that could potentially decrease performance when using many threads.
+        if (omp_get_thread_num() == 0 )
+          fprintf(stderr, STATUS_FORMAT,
+              finished_packets * omp_get_num_threads() * 100 / storage->no_of_packets,
+              finished_packets * omp_get_num_threads(),
+              storage->no_of_packets);
+#else
+        fprintf(stderr, STATUS_FORMAT, finished_packets*100/storage->no_of_packets, finished_packets, storage->no_of_packets);
```

Comment:
Well pep8 actually allows to omit spaces before and after operators it
looks better. But yes you should surround * with spaces unless it is a
pointer.
On 12 Feb 2016 15:19, "Wolfgang Kerzendorf" <notifications@github.com>
wrote:

> In tardis/montecarlo/src/cmontecarlo.c
> <https://github.com/tardis-sn/tardis/pull/485#discussion_r52739511>:
>
> >      {
> > +      ++finished_packets;
> > +      if ( finished_packets%100 == 0 ) {
> > +#ifdef WITHOPENMP
> > +        // WARNING: This only works with a static sheduler and gives an approximation of progress.
> > +        // The alternative would be to have a shared variable but that could potentially decrease performance when using many threads.
> > +        if (omp_get_thread_num() == 0 )
> > +          fprintf(stderr, STATUS_FORMAT,
> > +              finished_packets * omp_get_num_threads() * 100 / storage->no_of_packets,
> > +              finished_packets * omp_get_num_threads(),
> > +              storage->no_of_packets);
> > +#else
> > +        fprintf(stderr, STATUS_FORMAT, finished_packets*100/storage->no_of_packets, finished_packets, storage->no_of_packets);
>
> that is not PEP8 compatible (which applies to python, I know). Does GNU C
> suggest not having spaces between *?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tardis-sn/tardis/pull/485/files#r52739511>.
>